tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/EventsProcessor.scala

```diff
@@ -26,9 +28,9 @@ import org.apache.spark.scheduler._
 import org.apache.spark.sql.execution.ui.{SparkListenerDriverAccumUpdates, SparkListenerSQLAdaptiveExecutionUpdate, SparkListenerSQLAdaptiveSQLMetricUpdates, SparkListenerSQLExecutionEnd, SparkListenerSQLExecutionStart}
 
 /**
- * This object is to process all events and do validation in the end.
+ * This class is to process all events and do validation in the end.
  */
-object EventsProcessor extends Logging {
+class EventsProcessor(forQualification: Boolean = false) extends Logging {
```

Comment:
maybe make an enum to make it easier to add different use cases later

Reply:
we can do this later when we don't have so tight deadline, its functionally correct for now